plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,180 @@
+	Now           int32  `xmlrpc:"now"`
+	State         int16  `xmlrpc:"state"`
+	Statename     string `xmlrpc:"statename"`
+	StdoutLogfile string `xmlrpc:"stdout_logfile"`
+	StderrLogfile string `xmlrpc:"stderr_logfile"`
+	SpawnErr      string `xmlrpc:"spawnerr"`
+	ExitStatus    int8   `xmlrpc:"exitstatus"`
+	Pid           int32  `xmlrpc:"pid"`
+}
+
+type supervisorInfo struct {
+	StateCode int8   `xmlrpc:"statecode"`
+	StateName string `xmlrpc:"statename"`
+	Ident     string
+}
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+func (s *Supervisor) Description() string {
+	return "Gather info about processes state, that running under supervisor using its XML-RPC API"
+}
+
+func (s *Supervisor) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Supervisor) Gather(acc telegraf.Accumulator) error {
+	// API call to get information about all running processes
+	var rawProcessData []processInfo
+	err := s.rpcClient.Call("supervisor.getAllProcessInfo", nil, &rawProcessData)
+	if err != nil {
+		return fmt.Errorf("failed to get processes info: %v", err)
+	}
+
+	// API call to get information about instance status
+	var status supervisorInfo
+	err = s.rpcClient.Call("supervisor.getState", nil, &status)
+	if err != nil {
+		return fmt.Errorf("failed to get processes info: %v", err)
+	}
+
+	// API call to get identification string
+	err = s.rpcClient.Call("supervisor.getIdentification", nil, &status.Ident)
+	if err != nil {
+		return fmt.Errorf("failed to get instance identification: %v", err)
+	}
+
+	// Iterating through array of structs with processes info and adding fields to accumulator
+	for _, process := range rawProcessData {
+		processTags, processFields, err := s.parseProcessData(process, status)
+		if err != nil {
+			acc.AddError(err)
+			continue
+		}
+		acc.AddFields("supervisor_processes", processFields, processTags)
+	}
+	// Adding instance info fields to accumulator
+	instanceTags, instanceFields, err := s.parseInstanceData(status)
+	if err != nil {
+		return fmt.Errorf("failed to parse instance data: %v", err)
+	}
+	acc.AddFields("supervisor_instance", instanceFields, instanceTags)
+	return nil
+}
+
+func (s *Supervisor) parseProcessData(pInfo processInfo, status supervisorInfo) (map[string]string, map[string]interface{}, error) {
+	tags := map[string]string{
+		"process": pInfo.Name,
+		"group":   pInfo.Group,
+	}
+	fields := map[string]interface{}{
+		"uptime": pInfo.Now - pInfo.Start,
+		"state":  pInfo.State,
+	}
+	if s.fieldFilter.Match("pid") {
+		fields["pid"] = pInfo.Pid
+	}
+	if s.fieldFilter.Match("rc") {
+		fields["exitCode"] = pInfo.ExitStatus
+	}
+	splittedURL, err := beautifyServerString(s.Server)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse server string: %v", err)
+	}
+	tags["supervisor_id"] = status.Ident
+	tags["supervisor_host"] = splittedURL[0]
+	tags["supervisor_port"] = splittedURL[1]
```

Comment:
Could you rename supervisor_host to source and supervisor_port to port to match telegraf's convention?

Also since the metric name includes supervisor, let's simplify supervisor_id to id.

```suggestion
	tags["id"] = status.Ident
	tags["source"] = splittedURL[0]
	tags["port"] = splittedURL[1]
```

Reply:
Done